table_to_json.py

```diff
@@ -4,82 +4,97 @@
+                        temporary_table[idx, idxtd + column_shift] = single_td.text.strip()
+                    else:
+                        for j in range(rowspan_range):
+                            for i in range(colspan_range):
+                                while temporary_table[idx + j, idxtd + column_shift + i]:
+                                    column_shift += 1
+                                temporary_table[idx + j, idxtd + column_shift + i] = single_td.text.strip()
+                        column_shift += int(colspan_range) - 1
+        return temporary_table
 
-converted_tr = soup.find_all('tr')
-table_width = 0
 
+    # search for th in each matrix row to the left from td (is_row) or in each column above td
+    def __search_th(self, search_range, pos_td, is_row):
+        temp_name = ''
 
-#find width of the table
-for single_td in converted_tr[0].find_all('th'):
-     try:
-         table_width += int(single_td["colspan"])
-     except (ValueError, KeyError) as e:
-         table_width += 1
+        for s in range(search_range):
+            if is_row:
+                second_pos, first_pos = s, pos_td
+            else:
+                second_pos, first_pos = pos_td, s
 
+            if self.temporary_table_th[first_pos,  second_pos]:
+                temp_name = temp_name + self.temporary_table_th[first_pos,  second_pos] + '.'
 
-#add only td to table named temporary_table_td
-temporary_table_td = create_matrix('td')
+        return temp_name
 
-#add only th to table named temporary_table_th
-temporary_table_th = create_matrix('th')
 
-print('Matrix for <td>: \n', temporary_table_td)
-print('\nMatrix for <th>: \n', temporary_table_th)
+    def show_td_matrix(self):
+        print('Matrix for <td>: \n', self.temporary_table_td)
 
 
-temporary_dict = {}
-#change td matrix to dictionary with key from th rows.columns
-for i in range(len(converted_tr)):
-    for j in range(table_width):
-        if temporary_table_td[i, j]:
-            temporary_dict[(search_th(j, i, True) + search_th(i, j, False))[:-1]] = temporary_table_td[i, j]
+    def show_th_matrix(self):
+        print('\nMatrix for <th>: \n', self.temporary_table_th)
 
-jsonarray = json.dumps(temporary_dict, ensure_ascii=False)
 
-print('output json:')
-print(jsonarray)
\ No newline at end of file
+    def get_json(self):
+        #return json format
+        temporary_dict = {}
+
+        # change td matrix to dictionary with key from th rows.columns
+        for i in range(len(self.converted_tr)):
+            for j in range(self.table_width):
+                if self.temporary_table_td[i, j]:
+                    temporary_dict[(TableToJSON.__search_th(self, j, i, True) + TableToJSON.__search_th(self, i, j, False))[:-1]] = self.temporary_table_td[i, j]
+
+        jsonarray = json.dumps(temporary_dict, ensure_ascii=False)
+
+        return jsonarray
+
+
+if __name__ == "__main__":
+    print(TableToJSON.__doc__)
+    vivo = TableToJSON("vivo_ok.html")
```

Comment:
vivo_ok.html nie powinno być na sztywno, przekaż nazwę jako argument do pliku, podp.: sys.argv albo użyj całkiem zgrabnej biblioteki argparse https://docs.python.org/3/library/argparse.html
Wrzucanie takich rzeczy na sztywno nie jest dobre, bo ma to być kod uniwersalny, a nie do jednego pliku tylko